src/luxonis_ml/embeddings/ldf_emb.py

```diff
@@ -0,0 +1,292 @@
+        transform = transforms.Compose([
+            transforms.ToPILImage(),
+            transforms.Resize((224, 224)),  # Resize images to (224, 224) or any desired size
+            transforms.ToTensor(),
+            transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])  # Normalize images
+        ])
+
+    for i in range(0, len(new_payloads), emb_batch_size):
+        batch= new_payloads[i:i+emb_batch_size]
+        batch_img_paths = [payload["image_path"] for payload in batch]
+        
+        # Load, preprocess, and resize a batch of images
+        batch_images = [cv2.imread(img_path) for img_path in batch_img_paths]
+        batch_tensors = [transform(img) for img in batch_images]
+        batch_tensor = torch.stack(batch_tensors).cuda()
+
+        # Run the ONNX model on the batch
+        ort_inputs = {ort_session.get_inputs()[0].name: batch_tensor.cpu().numpy()}
+        ort_outputs = ort_session.run([output_layer_name], ort_inputs)
+        
+        # Append the embeddings from the batch to the new_embeddings list
+        batch_embeddings = ort_outputs[0].squeeze()
+        new_embeddings.extend(batch_embeddings.tolist())
+    
+    return new_embeddings
+
+def _batch_upsert(qdrant_client, new_embeddings, new_payloads, qdrant_batch_size = 64, qdrant_collection_name="mnist"):
+    """
+    Perform batch upserts of embeddings to Qdrant.
+    Args:
+        qdrant_client: Qdrant client instance.
+        new_embeddings: List of new embeddings.
+        new_payloads: List of new payloads.
+        qdrant_batch_size: Batch size for inserting into Qdrant.
+        qdrant_collection_name: Name of the Qdrant collection.
+    """
+    # Perform batch upserts to Qdrant
+    for i in range(0, len(new_embeddings), qdrant_batch_size):
+        batch_embeddings = new_embeddings[i:i+qdrant_batch_size]
+        batch_payloads = new_payloads[i:i+qdrant_batch_size]
+
+        points = [models.PointStruct(
+            id=payload["instance_id"],
+            vector=embedding,
+            payload=payload
+        ) for j, (embedding, payload) in enumerate(zip(batch_embeddings, batch_payloads))]
+
+        try:
+            qdrant_client.upsert(collection_name=qdrant_collection_name, points=points)
+            print(f"Upserted batch {i // qdrant_batch_size + 1} / {len(new_embeddings) // qdrant_batch_size + 1} of size {len(points)} to Qdrant.")
+        except Exception as e:
+            print(e)
+            print(f"Failed to upsert batch {i // qdrant_batch_size + 1} to Qdrant.")
+
+def generate_embeddings(luxonis_dataset, 
+                         ort_session, 
+                         qdrant_client, 
+                         qdrant_collection_name=DEFAULT_COLLECTION_NAME, 
+                         output_layer_name=DEFAULT_OUTPUT_LAYER_NAME,
+                         emb_batch_size=DEFAULT_EMB_BATCH_SIZE, 
+                         qdrant_batch_size=DEFAULT_QDRANT_BATCH_SIZE):
```

Comment:
Oh I see the global variables are only used here. It seems inconsistent with other functions though